### PR TITLE
Regex support with --table-list and --exclude-table-list in export/import data

### PR DESCRIPTION
**Pull request by priyanshi-yb**
This patch adds regex support in `table-list` and `exclude-table-list`, we can pass a regex in table list or exclude table list as e.g. -
```
--table-list 'emp_*'  #emp_ prefixed tables
--table-list 'abc."Cust_*"'  # "Cust_*"  tables in abc schema
```

### CODE REVIEW

File: yb-voyager/cmd/importData.go [hunk 1]
```diff
@@ -241,43 +241,69 @@ func discoverFilesToImport() []*ImportFileTask {
...
-	log.Infof("includeList: %v", includeList)
-	excludeList := utils.CsvStringToSlice(tconf.ExcludeTableList)
-	log.Infof("excludeList: %v", excludeList)
+
+	migInfo, err := LoadMigInfo(exportDir) //TODO: handle with msr.SourceDBConf
+	if err != nil {
+		utils.ErrExit("failed to load migration info: %s", err)
+	}
+	source.DBType = migInfo.SourceDBType
+	source.Schema = migInfo.SourceDBSchema
```
Comment (amit-yb) [inline yb-voyager/cmd/importData.go:250]: Sanyam might have merged his msr related changes. So you can use `msr.SourceDBConf`.

File: yb-voyager/cmd/importData.go [hunk 2]
```diff
@@ -241,43 +241,69 @@ func discoverFilesToImport() []*ImportFileTask {
...
-		return standardizeCaseInsensitiveTableNames(task.TableName)
+		defaultSourceSchema, _ := getDefaultSourceSchemaName() // err can be ignored as these table names will be qualified for non-public schemas
+		return standardizeCaseInsensitiveTableNames(task.TableName, defaultSourceSchema)
 	})
 	slices.Sort(allTables)
 	log.Infof("allTables: %v", allTables)
 
+	flattenTables := func(pattern string) []string {
+		result := lo.Filter(allTables, func(tableName string, _ int) bool {
+			matched, _ := filepath.Match(pattern, tableName)
```
Comment (amit-yb) [inline yb-voyager/cmd/importData.go:277]: Exit if `Match()` returns an error.

File: yb-voyager/cmd/importData.go [hunk 3]
```diff
@@ -241,43 +241,69 @@ func discoverFilesToImport() []*ImportFileTask {
...
+		for _, table := range tableList {
+			if err != nil && len(strings.Split(table, ".")) == 1 {
+				unqualifiedTables = append(unqualifiedTables, table)
+				continue
+			}
+			table = standardizeCaseInsensitiveTableNames(table, defaultSourceSchema)
+			result = append(result, flattenTables(table)...)
+		}
+		if len(unqualifiedTables) > 0 {
+			utils.PrintAndLog("Error: can not qualify table names %v in the %s list. \n%s", unqualifiedTables, listName, err.Error())
```
Comment (amit-yb) [inline yb-voyager/cmd/importData.go:297]: `"qualify following names from the %s list with schema-name: %s", listName, unqualifiedTables`

File: yb-voyager/cmd/importData.go [hunk 4]
```diff
@@ -295,11 +321,12 @@ func applyTableListFilter(importFileTasks []*ImportFileTask) []*ImportFileTask {
 	checkUnknownTableNames(excludeList, "exclude")
 
 	for _, task := range importFileTasks {
-		if len(includeList) > 0 && !slices.Contains(includeList, standardizeCaseInsensitiveTableNames(task.TableName)) {
+		defaultSourceSchema, _ := getDefaultSourceSchemaName()
```
Comment (amit-yb) [inline yb-voyager/cmd/importData.go:324]: Call this function only once at the beginning of `applyTableListFilter()`.

File: yb-voyager/cmd/importData.go [hunk 5]
```diff
@@ -295,11 +321,12 @@ func applyTableListFilter(importFileTasks []*ImportFileTask) []*ImportFileTask {
 	checkUnknownTableNames(excludeList, "exclude")
 
 	for _, task := range importFileTasks {
-		if len(includeList) > 0 && !slices.Contains(includeList, standardizeCaseInsensitiveTableNames(task.TableName)) {
+		defaultSourceSchema, _ := getDefaultSourceSchemaName() 
+		if len(includeList) > 0 && !slices.Contains(includeList, standardizeCaseInsensitiveTableNames(task.TableName, defaultSourceSchema)) {
```
Comment (amit-yb) [inline yb-voyager/cmd/importData.go:325]: Line too long.

Factor out following duplicated statement:
`tableName := standardizeCaseInsensitiveTableNames(task.TableName, defaultSchemaName)`
Pass `tableName` to the `slice.Contains()` calls.

File: yb-voyager/cmd/exportData.go [hunk 6]
```diff
@@ -330,34 +332,67 @@ func checkDataDirs() {
...
 
-func getDefaultSourceSchemaName() string {
+func getDefaultSourceSchemaName() (string, error) {
 	switch source.DBType {
 	case MYSQL:
-		return source.DBName
+		return source.DBName, nil
 	case POSTGRESQL, YUGABYTEDB:
-		return "public"
+		schemas := strings.Split(source.Schema, "|")
```
[the remark this answers is not in the document]
Comment (priyanshi-yb) [reply yb-voyager/cmd/exportData.go:340]: Actually, we only support passing comma-separated values in the schema for PG via voyager command,  this pipe character as a separator is what we are modifying internally for better representation for pg_dump/pg. 
```
func validateSourceSchema() {
	if source.Schema == "" {
		return
	}

	schemaList := utils.CsvStringToSlice(source.Schema)
	....
	case POSTGRESQL:
		// In PG, its supported to export more than one schema
		source.Schema = strings.Join(schemaList, "|") // clean and correct formatted for pg
	}
}
```

File: yb-voyager/cmd/exportData.go [hunk 7]
```diff
@@ -178,11 +179,12 @@ func getFinalTableColumnList() ([]*sqlname.SourceName, map[*sqlname.SourceName][
 	var tableList []*sqlname.SourceName
 	// store table list after filtering unsupported or unnecessary tables
 	var finalTableList, skippedTableList []*sqlname.SourceName
-	excludeTableList := extractTableListFromString(source.ExcludeTableList)
+	fullTableList := source.DB().GetAllTableNames()
+	excludeTableList := extractTableListFromString(fullTableList, source.ExcludeTableList, "exclude")
 	if source.TableList != "" {
-		tableList = extractTableListFromString(source.TableList)
+		tableList = extractTableListFromString(fullTableList, source.TableList, "include")
```
Comment (makalaaneesh) [inline yb-voyager/cmd/exportData.go:185]: @priyanshi-yb can you update the table-list (and exclude-table-list) argument help messages to include this pattern support?

Comment (priyanshi-yb) [reply yb-voyager/cmd/exportData.go:185]: yes, will do

File: yb-voyager/cmd/export.go [hunk 8]
```diff
@@ -243,10 +243,10 @@ func registerExportDataFlags(cmd *cobra.Command) {
 		"true - to disable progress bar during data export and stats printing during streaming phase (default false)")
 
 	cmd.Flags().StringVar(&source.ExcludeTableList, "exclude-table-list", "",
-		"list of tables to exclude while exporting data")
+		"comma separated list of tables names or regular expressions for table names to exclude while exporting data")
```
Comment (amit-yb) [inline yb-voyager/cmd/export.go:246]: Mention that `?` matches one char and `*` matches zero-or-more.

File: yb-voyager/cmd/import.go [hunk 9]
```diff
@@ -190,9 +190,9 @@ func registerImportDataFlags(cmd *cobra.Command) {
 	BoolVar(cmd.Flags(), &disablePb, "disable-pb", false,
 		"true - to disable progress bar during data import and stats printing during streaming phase (default false)")
 	cmd.Flags().StringVar(&tconf.ExcludeTableList, "exclude-table-list", "",
-		"list of tables to exclude while importing data")
+		"comma separated list of tables names or regular expressions for table names to exclude while importing data")
```
Comment (amit-yb) [inline yb-voyager/cmd/import.go:193]: Same as previous comment.

File: yb-voyager/cmd/importData.go [hunk 10]
```diff
@@ -237,12 +237,13 @@ func discoverFilesToImport() []*ImportFileTask {
...
-	migInfo, err := LoadMigInfo(exportDir) //TODO: handle with msr.SourceDBConf
+	msr, err := metaDB.GetMigrationStatusRecord()
 	if err != nil {
-		utils.ErrExit("failed to load migration info: %s", err)
+		utils.ErrExit("could not fetch migration status record: %w", err)
 	}
-	source.DBType = migInfo.SourceDBType
-	source.Schema = migInfo.SourceDBSchema
+	source.DBType = msr.SourceDBConf.DBType
+	source.Schema = msr.SourceDBConf.Schema
```
Comment (amit-yb) [inline yb-voyager/cmd/importData.go:245]: How about:
`source = *msr.SourceDBConf`

**Review by amit-yb — APPROVED** [review]
LGTM with a few minor comments. Address them and proceed to merge the changes.

**Pull request merged**